### PR TITLE
(CL): Improve `GetTickLiquidityForRange` and `GetTickLiquidityNetInDirection` query performance by using iterators

### PR DESCRIPTION
**Pull request by mattverse**
Closes: https://github.com/osmosis-labs/osmosis/issues/4704

## What is the purpose of the change
Improves performance of`GetTickLiquidityForRange` query by using prefix iterators directly.

## Brief Changelog

Improve `GetTickLiquidityForRange` query performance by using iterators

## Testing and Verifying

Existing tests passes

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Review by p0mvn — COMMENTED** [review]
Nice work.

Have a couple of questions/comments:
- Can We rename the function to the `GetTickLiquidityForFullRange` because we do not allow specifying lower and upper bounds?
- The godoc needs to be updated
- Can this iterator improvement algorithm be also applied to the query introduce in: https://github.com/osmosis-labs/osmosis/pull/4714

**Comment by mattverse** [comment]
Hey @p0mvn  thanks for the review, while I was resolving your suggestions, I actually came across the fundamental question  of needing to improve query performance via this PR.

Right now, the code operates based on NextInitialized tick. If we take a look at the implementation,
https://github.com/osmosis-labs/osmosis/blob/8ec7dadbaf/x/concentrated-liquidity/internal/swapstrategy/one_for_zero.go#L167-L192 we can see that it uses iterators any way. Thus I came to a conclusion that this PR is not necessarily needed in the first place. Wdyt?

File: x/concentrated-liquidity/tick.go [hunk 1]
```diff
@@ -220,32 +222,49 @@ func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]quer
...
+	for ; iterator.Valid(); iterator.Next() {
+		tickIndex, err := types.TickIndexFromBytes(iterator.Key())
+		if err != nil {
+			return []query.LiquidityDepthWithRange{}, err
 		}
 
-		tick, err := k.getTickByTickIndex(ctx, poolId, nextTick)
+		keyTick := types.KeyTick(poolId, tickIndex)
+		tickStruct := model.TickInfo{}
+		found, err := osmoutils.Get(store, keyTick, &tickStruct)
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/concentrated-liquidity/tick.go:247]: Fixed it to use directly using iterator.Value()

**Comment by mattverse** [comment]
Hey @p0mvn thanks for the review, in the most recent commit I have made the following changes,
- Add `GetTickLiquidityNetInDirection` improvement
- Godoc updated
- Rename `GetTickLiquidityForFullRange`
- Address and resolve other comments made in previous iteration of review, 
Please take a look!

File: x/concentrated-liquidity/tick.go [hunk 2]
```diff
@@ -182,8 +186,8 @@ func GetMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne sdk.Int) (minTic
 	return math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
 }
 
-// GetTickLiquidityForRangeInBatches returns an array of liquidity depth within the given range of lower tick and upper tick.
-func (k Keeper) GetTickLiquidityForRange(ctx sdk.Context, poolId uint64) ([]query.LiquidityDepthWithRange, error) {
+// GetTickLiquidityForFullRangeInBatches returns an array of liquidity depth for all ticks existing from min tick ~ max tick.
```
Comment (p0mvn) [inline x/concentrated-liquidity/tick.go:189]:
```suggestion
// GetTickLiquidityForFullRange returns an array of liquidity depth for all ticks existing from min tick ~ max tick.
```

**Review by p0mvn — APPROVED** [review]
Awesome, LGTM!

**Pull request merged**